src/lib/mongoBackend/MongoCommonUpdate.cpp

```diff
@@ -1886,17 +1886,37 @@ static bool processSubscriptions
     if (notificationSent)
     {
       long long rightNow = getCurrentTime();
-
+      BSONObj query  = BSON("_id" << OID(mapSubId));
+      BSONObj update;
       //
       // If broker running without subscription cache, put lastNotificationTime and count in DB
       //
       if (subCacheActive == false)
       {
-        BSONObj query  = BSON("_id" << OID(mapSubId));
-        BSONObj update = BSON("$set" <<
-                              BSON(CSUB_LASTNOTIFICATION << rightNow) <<
-                              "$inc" << BSON(CSUB_COUNT << (long long) 1));
+        BSONObj subOrig;
+        std::string newErr;
+        collectionFindOne(getSubscribeContextCollectionName(tenant), BSON("_id" << OID(mapSubId)), &subOrig, &newErr);
+        std::string status;
+        if (!subOrig.isEmpty())
+        {
+          if (subOrig.hasField(CSUB_STATUS))
+          {
+            status = getStringFieldF(subOrig, CSUB_STATUS);
+          }
+        }
 
+        // Update the value of status (in case of oneshot) in DB when broker is running without subscription cache
+        if (status == STATUS_ONESHOT)
+        {
+          update = BSON("$set" << BSON(CSUB_LASTNOTIFICATION << rightNow) <<
```

Comment:
Maybe I didn't may point clear. I'll try again :)

What you have now leads to (at MongoDB level):

```
{ $set: { lastNotification: ... }, $set: { status: ... }, $inc: { } }
```

Note you are duplicating a key at root level. Although this may work, note it doesn't compliant with the [MongoDB $set operator documentation](https://docs.mongodb.com/manual/reference/operator/update/set) which states:

```
{ $set: { <field1>: <value1>, ... } }
```

Thus, you should achieve

```
{ $set: { lastNotification: ..., status: ... }, $inc: { } }
```

As far as I understand the following statement will achieve that:



It would be more correct to group both set statements. If I remember correctly it would be like:

```
      update = BSON("$set" << BSON(CSUB_LASTNOTIFICATION << rightNow << CSUB_STATUS << STATUS_INACTIVE) <<
                    "$inc" << BSON(CSUB_COUNT << (long long) 1));
```

Reply:
fixed in 00406ae